File: packages/python/examples/submit_operation_event.py
import asyncio

from seclab_suite_runtime import (
    OperationEvent,
    OperationImpact,
    OperationOutcome,
    RuntimeClient,
)


async def main() -> None:
    client = await RuntimeClient.from_environment("operation-logs.write")
    try:
        await client.submit_operation_event(
            OperationEvent(
                event_code="task_submitted",
                zh_cn="提交任务",
                en_us="Submit task",
                outcome=OperationOutcome.SUCCESS,
                impact=OperationImpact.INFO,
            )
        )
    finally:
        await client.aclose()


asyncio.run(main())

File: packages/rust/examples/submit_operation_event.rs
use seclab_suite_runtime::{OperationEvent, OperationImpact, OperationOutcome, RuntimeClient};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = RuntimeClient::from_environment("operation-logs.write").await?;
    let event = OperationEvent::builder(
        "task_submitted",
        "提交任务",
        "Submit task",
        OperationOutcome::Success,
        OperationImpact::Info,
    )
    .build()?;
    client.submit_operation_event(&event).await?;
    Ok(())
}
